Sources/BraintreePayPal/BTPayPalClient.swift

```diff
@@ -230,24 +231,27 @@ import BraintreeDataCollector
             self.apiClient.post(request.hermesPath, parameters: request.parameters(with: configuration)) { body, response, error in
                 if let error = error as? NSError {
                     if error.code == BTCoreConstants.networkConnectionLostCode {
-                        self.apiClient.sendAnalyticsEvent("ios.paypal.tokenize.network-connection.failure")
+                        self.apiClient.sendAnalyticsEvent(BTPayPalAnalytics.tokenizeNetworkConnectionFailed)
                     }
 
                     guard let jsonResponseBody = error.userInfo[BTCoreConstants.jsonResponseBodyKey] as? BTJSON else {
+                        self.apiClient.sendAnalyticsEvent(BTPayPalAnalytics.tokenizeFailed)
                         completion(nil, error)
                         return
                     }
 
                     let errorDetailsIssue = jsonResponseBody["paymentResource"]["errorDetails"][0]["issue"]
                     var dictionary = error.userInfo
                     dictionary[NSLocalizedDescriptionKey] = errorDetailsIssue
+                    self.apiClient.sendAnalyticsEvent(BTPayPalAnalytics.tokenizeFailed)
                     completion(nil, BTPayPalError.httpPostRequestError(dictionary))
```

Comment:
Based on the surrounding code above this line:
```swift
guard let jsonResponseBody = error.userInfo[BTCoreConstants.jsonResponseBodyKey] as? BTJSON else {
    completion(nil, error)
    return
}

jsonResponseBody["paymentResource"]["errorDetails"][0]["issue"]
```
It looks like if the error _isn't_ of a certain structure, we just return the error. Otherwise, if the user info has a `jsonResponseBodyKey` we extract the error out from the `paymentResource`. There is a test `testTokenizePayPalAccount_whenPostRequestContainsError_callsBackWithError` that shows the format of what we may get back, so we mainly do it to make the error more readable for merchants.